### ISSUE TITLE
homekit group item exceptions

### ISSUE THREAD

**Issue by digitaldan** [openhab/openhab2-addons#986]
Hi @beowulfe , I have just started using the homekit binding and run into an issue.  I have many Group:Switch and Group:Dimmer Items which represent rooms in my house, for example

`Group:Switch:OR(ON, OFF)	Group_Second_Floor  "Second Floor [%s]"	  ["Switchable"] `

This group is treated like a switch in OH (its baseItem is a Switch).  In the homekit binding this throws an exception:
```2016-05-30 10:58:44.694 [ERROR] [lfe.hap.impl.connections.HttpSession] - Could not handle request
java.lang.ClassCastException: org.eclipse.smarthome.core.items.GroupItem cannot be cast to org.eclipse.smarthome.core.library.items.SwitchItem
	at org.openhab.io.homekit.internal.accessories.AbstractHomekitLightbulbImpl.getLightbulbPowerState(AbstractHomekitLightbulbImpl.java:37)[224:org.openhab.io.homekit:2.0.0.201605301758]```

Let me know if I can be of assistance, thanks.

**Comment by beowulfe** [comment]
Interesting. I haven't used that capability before, and wasn't really aware of it.

@kaikreuzer (or anyone else) - is there a way to get a SwitchItem from the org.eclipse.smarthome.core.items.GroupItem? The comments on getBaseItem() indicate that won't work.

I dug into ESH a bit, but couldn't find a great way to accomplish this short of littering every HomeKit accessory implementation with: if (item instanceof GroupItem)..... which I'd like to avoid if possible.

**Comment by digitaldan** [comment]
@beowulfe I have a branch that I think fixes this, i'll open a PR for review.  Unfortunately I'm away from my home and and can't test it until Thurs PDT ;-( But at least you can review it.  I did have to add a couple of instanceof checks, but in the end it was very minor.

**Comment by beowulfe** [comment]
This issue can be closed - the merge of #992 resolved it.
